### PR TITLE
`PyASTBridge` : add more numpy constants to `visit_Attribute`

### PR DESCRIPTION
**Pull request by freifrauvonbleifrei**
### Description

related to #1734 , such that Euler's number can be resolved (and an error is thrown if no named math constant is found).

### CODE REVIEW

**Comment by freifrauvonbleifrei** [comment]
@annagrin feel free to have a look, it's just a small change ;)

File: python/tests/display/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+# # This source code and the accompanying materials are made available under #
+# the terms of the Apache License 2.0 which accompanies this distribution. #
+# ============================================================================ #
+
+add_test(
```
Comment (amccaskey) [inline python/tests/display/CMakeLists.txt:8]: someone else can correct me if I'm wrong, but I'm not sure we need this CMakeLists file. The CI already globs up all  pytests in the `tests/*` folder so this new `test_draw.py` file should be run already. See here https://github.com/NVIDIA/cuda-quantum/blob/main/.github/workflows/test_in_devenv.yml#L209

Comment (freifrauvonbleifrei) [reply python/tests/display/CMakeLists.txt]: This makes sense. I had tried out a weird workaround for running this test apparently -- deleted!

File: python/tests/visualization/test_draw.py [hunk 2]
```diff
@@ -24,13 +24,9 @@ def test_draw():
 
     @cudaq.kernel
     def bar(qvec: cudaq.qview):
-        # FIXME https://github.com/NVIDIA/cuda-quantum/issues/1734
-        # rx(np.e, qvec[0])
-        rx(2.71828182845904523536028, qvec[0])
+        rx(np.e, qvec[0])
```
Comment (annagrin) [inline python/tests/visualization/test_draw.py:27]: Can we add a test for `euler_gamma` as well? and maybe a negative test for something that is not supported?

Comment (khalatepradnya) [reply python/tests/visualization/test_draw.py:27]: Addressed in 96d4ad8310c72779d7a22a262c8c24f026d21c46

**Review by annagrin — APPROVED** [review]
Thanks Pradnya, LGTM!
